keeper/query_server_vault_v2.go

```diff
@@ -0,0 +1,1417 @@
+	err = q.IterateVaultsV2Withdrawals(ctx, func(id uint64, request vaultsv2.WithdrawalRequest) (bool, error) {
+		// Filter by user
+		if request.Requester != req.Address {
+			return false, nil
+		}
+
+		// Determine status
+		status := "PENDING"
+		if false {
+			if sdkmath.ZeroInt().IsPositive() {
+				if false {
+					status = "CLAIMED"
+				} else {
+					status = "CLAIMABLE"
+				}
+			} else {
+				status = "PROCESSING"
+			}
+		}
+
+		item := vaultsv2.WithdrawalStatusItem{
+			RequestId:       strconv.FormatUint(id, 10),
+			Amount:          request.WithdrawAmount.String(),
+			PrincipalAmount: request.WithdrawAmount.String(),
+			YieldAmount:     sdkmath.ZeroInt().String(),
+			FulfilledAmount: sdkmath.ZeroInt().String(),
+			Status:          status,
+			Timestamp:       request.RequestTime,
+		}
+
+		if false {
+			item.ClaimableAt = &time.Time{}
+		}
+		if false {
+			item.ClaimedAt = &time.Time{}
+		}
+
+		withdrawals = append(withdrawals, item)
+
+		// Accumulate totals
+		var err error
+		if status == "PENDING" || status == "PROCESSING" {
+			totalPending, err = totalPending.SafeAdd(request.WithdrawAmount)
+			if err != nil {
+				return true, errors.Wrap(err, "unable to accumulate pending")
+			}
+		} else if status == "CLAIMABLE" {
+			totalClaimable, err = totalClaimable.SafeAdd(sdkmath.ZeroInt())
+			if err != nil {
+				return true, errors.Wrap(err, "unable to accumulate claimable")
+			}
+		} else if status == "CLAIMED" {
+			totalClaimed, err = totalClaimed.SafeAdd(sdkmath.ZeroInt())
+			if err != nil {
+				return true, errors.Wrap(err, "unable to accumulate claimed")
+			}
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to iterate withdrawals")
+	}
+
+	return &vaultsv2.QueryUserWithdrawalsResponse{
+		Withdrawals:    withdrawals,
+		TotalPending:   totalPending.String(),
+		TotalClaimable: totalClaimable.String(),
+		TotalClaimed:   totalClaimed.String(),
+	}, nil
+}
+
+func (q queryServerV2) UserBalance(ctx context.Context, req *vaultsv2.QueryUserBalanceRequest) (*vaultsv2.QueryUserBalanceResponse, error) {
+	if req == nil {
+		return nil, errors.Wrap(types.ErrInvalidRequest, "request cannot be nil")
+	}
+
+	// Validate address
+	userAddr, err := q.address.StringToBytes(req.Address)
+	if err != nil {
+		return nil, errors.Wrapf(types.ErrInvalidRequest, "invalid address: %s", req.Address)
+	}
+
+	// Get user position
+	position, found, err := q.GetVaultsV2UserPosition(ctx, userAddr)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to fetch user position")
+	}
+	if !found {
+		return nil, errors.Wrap(types.ErrInvalidRequest, "user position not found")
+	}
+
+	if err != nil {
+		return &vaultsv2.QueryUserBalanceResponse{
+			DepositAmount:  sdkmath.ZeroInt().String(),
+			AccruedYield:   sdkmath.ZeroInt().String(),
+			TotalValue:     sdkmath.ZeroInt().String(),
+			UnrealizedGain: sdkmath.ZeroInt().String(),
+			LockedAmount:   sdkmath.ZeroInt().String(),
+		}, nil
+	}
```

Comment:
Same as earlier; error was already handled, I think you meant to return the empty response on line 1007